sql/parser/type_check.go

```diff
@@ -444,10 +444,26 @@ func typeCheckCastExpr(expr *CastExpr) (Datum, error) {
 			return DummyString, nil
 		}
 
-	// TODO(pmattis): unimplemented.
-	case *DecimalType:
 	case *DateType:
+		switch dummyExpr {
+		case DummyDate, DummyString:
+			return DummyDate, nil
+		}
+
 	case *TimestampType:
+		switch dummyExpr {
+		case DummyTimestamp, DummyString:
+			return DummyTimestamp, nil
+		}
+
+	case *IntervalType:
+		switch dummyExpr {
+		case DummyInterval, DummyString:
```

Comment:
probably should also allow DInt and perhaps but I'll get to it when we pass an integer as a placeholder in the placeholder change.